### PR TITLE
Remove isinstance experimental feature warning

### PR DESCRIPTION
**Pull request by guilhermeleobas**
As title.

### CODE REVIEW

**Comment by stuartarchibald** [comment]
Thanks for the patch. I think the docs also need an update, here:
https://github.com/numba/numba/blob/d44573b43dec9a7b66e9a0d24ef8db94c3dc346c/docs/source/reference/pysupported.rst?plain=1#L920-L922
and here:
https://github.com/numba/numba/blob/d44573b43dec9a7b66e9a0d24ef8db94c3dc346c/docs/source/reference/pysupported.rst?plain=1#L939

**Comment by guilhermeleobas** [comment]
@stuartarchibald, done!

File: numba/tests/test_builtins.py [hunk 1]
```diff
@@ -1387,7 +1387,7 @@ def foo(x):
             self.assertEqual(got, expected)
 
     @TestCase.run_test_in_subprocess
-    def test_experimental_warning(self):
+    def test_no_experimental_warning(self):
```
Comment (stuartarchibald) [inline numba/tests/test_builtins.py:1390]: Now CI has run and this change has been effectively confirmed... what do you think to just deleting this whole test? I don't think there's a need to check that there is no warning raised here any more than in any other piece of code where no warning is expected?

Comment (guilhermeleobas) [reply numba/tests/test_builtins.py:1390]: Done!

Comment (guilhermeleobas) [reply numba/tests/test_builtins.py:1390]: Oops, my bad!

**Comment by guilhermeleobas** [comment]
@gmarkall, could you take another look at this PR once you have some free cycles to spare?

**Review by gmarkall — COMMENTED** [review]
The changes look good to me, but should this also undo the changes from #8947 which ought to be un-needed now?

**Comment by guilhermeleobas** [comment]
Thanks for the review

**Pull request merged**